chemprop/uncertainty/uncertainty_predictor.py

```diff
@@ -0,0 +1,1014 @@
+            dropout_sampling_size=dropout_sampling_size,
+            individual_ensemble_predictions=individual_ensemble_predictions,
+            spectra_phase_mask=spectra_phase_mask,
+        )
+        self.label = "classification_uncal_confidence"
+
+    def raise_argument_errors(self):
+        super().raise_argument_errors()
+        if self.dataset_type not in ["classification", "multiclass"]:
+            raise ValueError(
+                "Classification output uncertainty method must be used with dataset types classification or multiclass."
+            )
+
+    def calculate_predictions(self):
+        for i, (model, scaler_list) in enumerate(
+            tqdm(zip(self.models, self.scalers), total=self.num_models)
+        ):
+            (
+                scaler,
+                features_scaler,
+                atom_descriptor_scaler,
+                bond_feature_scaler,
+            ) = scaler_list
+            if (
+                features_scaler is not None
+                or atom_descriptor_scaler is not None
+                or bond_feature_scaler is not None
+            ):
+                self.test_data.reset_features_and_targets()
+                if features_scaler is not None:
+                    self.test_data.normalize_features(features_scaler)
+                if atom_descriptor_scaler is not None:
+                    self.test_data.normalize_features(
+                        atom_descriptor_scaler, scale_atom_descriptors=True
+                    )
+                if bond_feature_scaler is not None:
+                    self.test_data.normalize_features(
+                        bond_feature_scaler, scale_bond_features=True
+                    )
+
+            preds = predict(
+                model=model,
+                data_loader=self.test_data_loader,
+                scaler=scaler,
+                return_unc_parameters=False,
+            )
+            if i == 0:
+                sum_preds = np.array(preds)
+                if self.individual_ensemble_predictions:
+                    individual_preds = np.expand_dims(np.array(preds), axis=-1)
+                if model.train_class_sizes is not None:
+                    self.train_class_sizes = [model.train_class_sizes]
+            else:
+                sum_preds += np.array(preds)
+                if self.individual_ensemble_predictions:
+                    individual_preds = np.append(individual_preds, np.expand_dims(preds, axis=-1), axis=-1)
+                if model.train_class_sizes is not None:
+                    self.train_class_sizes.append(model.train_class_sizes)
+
+        self.uncal_preds = (sum_preds / self.num_models).tolist()
+        self.uncal_confidence = self.uncal_preds
+        if self.individual_ensemble_predictions:
+            self.individual_preds = individual_preds.tolist()
+
+    def get_uncal_output(self):
+        return self.uncal_confidence
+
+
+def uncertainty_predictor_builder(
```

Comment:
```suggestion
def build_uncertainty_predictor(
```
smallest of nitpicks but the terminology isn't right IMO. a function *does* something whereas a class is a either a *doer* or *container* of things. The naming of this function implies that the return value is a *doer*. Rather, the function *does* something by *building* an uncertainty predictor. There are other functions that are named like this but I focused on this one. You can ignore this comment because it's so minor, but figured I would bring it up.

Reply:
On board with this change. I named it like it was a class rather than a function. Guess I was in a class state of mind when I was writing these since I was doing it alongside the classes.